### PR TITLE
resolves #48: add week details subtitle

### LINKED ISSUE #48: Improve week view: Mention year(s) and month(s) of shown week
see also: https://github.com/ohwgiles/focal/pull/46

### PR DESCRIPTION
**Pull request by kstenschke**
Please feel free to adjust the formatting, e.g. to use month abbreviations instead of full month names or change the order of day / month / year. Sooner or later, i think, localization should take care of these formats.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -150,13 +151,39 @@ static void create_calendars(FocalApp* fm)
...
+{
+	WeekView* wv = FOCAL_WEEK_VIEW(fm->weekView);
+
+	struct tm day_start = week_view_get_start(wv);
+	char start[24];
+	strftime(start, sizeof(start), "%e. %B %G", &day_start);
+
+	struct tm day_end = week_view_get_end(wv);
+	char end[24];
+	day_end.tm_hour -= 2;
```
Comment (ohwgiles) [inline src/main.c]: The intention of this line isn't clear. If it's really necessary, a comment explaining it should be added

File: src/week-view.c [hunk 2]
```diff
@@ -565,7 +565,30 @@ void week_view_add_calendar(WeekView* wv, Calendar* cal)
 	gtk_widget_queue_draw((GtkWidget*) wv);
 }
 
-int week_view_get_current_week(WeekView* wv)
+struct tm toLocaltime(time_t* timestamp)
```
Comment (ohwgiles) [inline src/week-view.c]: Please use snake case for function names. Also, local module functions should be `static`

File: src/week-view.h [hunk 3]
```diff
@@ -28,7 +28,9 @@ void week_view_remove_event(WeekView* wv, Event* vevent);
 void week_view_focus_event(WeekView* wv, Event* event);
 void week_view_add_calendar(WeekView* widget, Calendar* cal);
 void week_view_remove_calendar(WeekView* wv, Calendar* cal);
-int week_view_get_current_week(WeekView* wv);
+struct tm week_view_get_start(WeekView* wv);
+struct tm week_view_get_end(WeekView* wv);
```
Comment (ohwgiles) [inline src/week-view.h]: `struct tm` is more complex than `time_t` and isn't the natural storage format used by `WeekView`. Perhaps these functions should just return the `time_t` and the caller should take care of the conversion if it needs it?

File: src/main.c [hunk 4]
```diff
@@ -253,6 +280,8 @@ static void open_prefs_dialog(GSimpleAction* simple, GVariant* parameter, gpoint
 		g_key_file_free(kf);
 		// update view
 		week_view_set_day_span(FOCAL_WEEK_VIEW(fm->weekView), fm->prefs.week_start_day, fm->prefs.week_end_day);
+
+		g_signal_emit_by_name(fm->weekView, "date-range-changed");
```
Comment (ohwgiles) [inline src/main.c]: Classes should emit their own signals. This should be moved inside `week_view_set_day_span` in the same fashion as used by e.g. [week_view_next](https://github.com/ohwgiles/focal/blob/6ab46231d666e19c8c41eb954337b472a52f32fd/src/week-view.c#L622)

File: src/main.c [hunk 5]
```diff
@@ -150,13 +151,39 @@ static void create_calendars(FocalApp* fm)
...
+
+	int week_num = week_view_get_week(wv);
 
-	gtk_window_set_title(GTK_WINDOW(fm->mainWindow), week_title);
+	char title[8];
+	snprintf(title, 8, "Week %d", week_num);
+
+	gtk_window_set_title(GTK_WINDOW(fm->mainWindow), title);
+
+	update_header_subtitle(fm);
```
Comment (ohwgiles) [inline src/main.c]: If there is never any situation where `update_header_subtitle` can be called independently of `update_window_title` then I prefer to use a single function (with comments if necessary marking the title and subtitle parts)

**Comment by kstenschke** [comment]
Thank you for the feedback, the latest commit should fix the mentioned issues.

**Comment by ohwgiles** [comment]
I'm on Arch Linux with gcc 8.2 and I get this compile error:

```
../src/main.c:177:33: error: ‘%s’ directive output may be truncated writing up to 39 bytes into a region of size between 36 and 75 [-Werror=format-truncation=]
  snprintf(subtitle, 80, "%s – %s", start, end);
                                 ^~          ~~~
../src/main.c:177:2: note: ‘snprintf’ output between 6 and 84 bytes into a destination of size 80
  snprintf(subtitle, 80, "%s – %s", start, end);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I'll add a fix to this PR

**Pull request merged**